src/main/scala/RocketChip.scala

```diff
@@ -271,25 +274,19 @@ class Uncore(implicit val p: Parameters) extends Module
   }
 
   def makeBootROM()(implicit p: Parameters) = {
-    val rom = java.nio.ByteBuffer.allocate(32)
-    rom.order(java.nio.ByteOrder.LITTLE_ENDIAN)
+    val romfile = Source.fromFile(p(BootROMFile))
+    val romdata = romfile.getLines().map(Integer.parseInt(_, 16))
+    val rom = ByteBuffer.allocate(romdata.size * 4)
+
+    rom.order(ByteOrder.LITTLE_ENDIAN)
+    rom.asIntBuffer().put(romdata.toArray)
 
     // for now, have the reset vector jump straight to memory
     val resetToMemDist = p(GlobalAddrMap)("mem").start - p(ResetVector)
     require(resetToMemDist == (resetToMemDist.toInt >> 12 << 12))
     val configStringAddr = p(ResetVector).toInt + rom.capacity
 
-    // This boot ROM doesn't know about any boot devices, so it just spins,
-    // waiting for the debugger to load a program and change the PC.
-    rom.putInt(0x0000006f)                        // loop forever
-    rom.putInt(0)                                 // reserved
-    rom.putInt(0)                                 // reserved
-    rom.putInt(configStringAddr)                  // pointer to config string
-    rom.putInt(0)                                 // default trap vector
-    rom.putInt(0)                                 //   ...
-    rom.putInt(0)                                 //   ...
-    rom.putInt(0)                                 //   ...
-
+    rom.putInt(12, configStringAddr)
```

Comment:
Maybe we should assert that (romdata.size < 4 || romdata(3) == 0) here, to make sure the boot ROM writer doesn't accidentally put some useful data where the config string address ends up?